_posts/2020-04-27-project.md

```diff
@@ -17,17 +17,48 @@ The aim of the project is to determine how real-time capabilities can be brought
   <img width="300" height="80" src="https://raw.githubusercontent.com/master-coro/rtros-documentation/master/img/ros-logo.png">
 </p>
 
-## A self-explanatory example
-
 ## What's hard real-time computing ?
 
 - It's about determinism, not performance.
 - Correct computation delivered at the correct time
 - Failure to respond is as bad as a wrong response
+- Missing a deadline is considered a system **failure**.
+- It is mandatory for **safety** or **mission-critical** systems.
+
+<p align="center">
+  <img width="500" height="400" src="img/rt_constraints.png">
+</p>
+
+On the top figure, each bar represent a task and the time at which it's executed. The red line represents a deadline. If a task is executed after the deadline, it is considered - in hard real-time systems - a failure. It has a different behaviour for firm or soft real-time systems. For firm constraints, a task that was executed after the deadline will not cause a system failure, but its behavior (i.e. computations) won't be considered. For soft constraints, the behavior will be considered with a non-critical warning.
+
+## Why do we need it ?
```

Comment:
The example is spot on. Especially the focus on the design aspect of things definitely helps a lot